sim/midas/src/main/cc/bridges/fased_memory_timing_model.cc

```diff
@@ -63,24 +63,32 @@ FASEDMemoryTimingModel::FASEDMemoryTimingModel(
 
   std::string suffix = "_" + std::to_string(modelno);
   for (auto &arg : args) {
-    if (arg.find("+mm_") == 0 && arg.find(suffix) != std::string::npos) {
-      auto sub_arg = arg.substr(4);
-      size_t delimit_idx = sub_arg.find("=");
-      size_t suffix_idx = sub_arg.find(suffix);
-      std::string key = sub_arg.substr(0, suffix_idx).c_str();
-
-      // This is the only nullary plusarg supported by fased
-      // All other plusargs are key-value pairs that will be written to the
-      // bridge module
-      if (key == std::string("useHardwareDefaultRuntimeSettings")) {
-        require_all_runtime_settings = false;
-      } else if (suffix_idx == std::string::npos) {
-        throw std::runtime_error("[FASED] unknown nullary plusarg: " + key);
-      } else {
-        int value = std::stoi(sub_arg.substr(delimit_idx + 1).c_str());
-        model_configuration[key] = value;
-      }
+    // Find arguments: +mm_{name}{index}={value}
+    if (arg.find("+mm_") != 0 || arg.find(suffix) == std::string::npos) {
+      continue;
+    }
+
+    auto sub_arg = arg.substr(4);
+    size_t delimit_idx = sub_arg.find("=");
+    size_t suffix_idx = sub_arg.find(suffix);
+    std::string key = sub_arg.substr(0, suffix_idx).c_str();
+
+    // This is the only nullary plusarg supported by fased
+    if (key == std::string("useHardwareDefaultRuntimeSettings")) {
```

Comment:
So if using the HW defaults is going to be the default, shouldn't this be renamed to "imposeCompleteConfiguration" or something similar, and the member name updated?

Reply:
It would be useful to clean this up, but we should have a discussion first, as I feel this ties into the stuff we want to do with argument parsing.